bindings/python/ethash/__init__.py

```diff
@@ -1,21 +1,39 @@
 # ethash: C/C++ implementation of Ethash, the Ethereum Proof of Work algorithm.
 # Copyright 2019 Pawel Bylica.
+# Copyright 2023 Sam Wilson.
 # Licensed under the Apache License, Version 2.0.
 
-from _ethash import ffi, lib
+from _ethash import ffi, lib  # type: ignore
 
+from typing import TYPE_CHECKING, Tuple
 
-def keccak_256(data):
+from collections.abc import Sized
+
+if TYPE_CHECKING:
+    from typing import Protocol
+
+    from _typeshed import WriteableBuffer, ReadableBuffer
+
+    class SizedWriteableBuffer(Protocol, WriteableBuffer, Sized):
+        pass
+
+    class SizedReadableBuffer(Protocol, ReadableBuffer, Sized):
+        pass
+
+
+def keccak_256(data: "SizedWriteableBuffer") -> bytes:
     hash = lib.ethash_keccak256(ffi.from_buffer(data), len(data))
     return ffi.unpack(hash.str, len(hash.str))
 
 
-def keccak_512(data):
+def keccak_512(data: "SizedWriteableBuffer") -> bytes:
     hash = lib.ethash_keccak512(ffi.from_buffer(data), len(data))
     return ffi.unpack(hash.str, len(hash.str))
 
 
-def hash(epoch_number, header_hash, nonce):
+def hash(
```

Comment:
*nit unrelated to PR:* `hash` is a reserved word in Python. This might be better as a different name in a next major version of the library. If unbothered by this, feel free to ignore this comment.

Reply:
I'm not sure this is worth changing now because this is API break (although I'm not sure many people is using it) and I have no good ideas for name.